pkg/controller/association/conf.go

```diff
@@ -103,6 +108,41 @@ func ElasticsearchAuthSettings(c k8s.Client, association commonv1.Association) (
 	return assocConf.AuthSecretKey, string(data), nil
 }
 
+// AllowVersion returns true if the given resourceVersion is lower or equal to the associations' versions.
+// For example: Kibana in version 7.8.0 cannot be deployed if its Elasticsearch association reports version 7.7.0.
+// Referenced resources version is parsed from the association conf annotation.
+func AllowVersion(resourceVersion version.Version, associated commonv1.Associated, logger logr.Logger, recorder record.EventRecorder) bool {
+	for _, assoc := range associated.GetAssociations() {
+		assocRef := assoc.AssociationRef()
+		if !assocRef.IsDefined() {
+			// no association specified, move on
+			continue
+		}
+		if assoc.AssociationConf() == nil || assoc.AssociationConf().Version == "" {
+			// no conf reported yet, this may be the initial resource creation
+			logger.Info("Delaying version deployment since the version of an associated resource is not reported yet",
+				"version", resourceVersion, "ref_namespace", assocRef.Namespace, "ref_name", assocRef.Name)
+			return false
+		}
+		refVer, err := version.Parse(assoc.AssociationConf().Version)
+		if err != nil {
+			logger.Error(err, "Invalid version found in association conf", "association_version", assoc.AssociationConf().Version)
```

Comment:
Nit: for clarity and consistency.
```suggestion
			logger.Error(err, "Invalid version found in association configuration", "association_version", assoc.AssociationConf().Version)
```